bot/seasons/valentines/movie_generator.py

```diff
@@ -0,0 +1,60 @@
+import logging
+import random
+from os import environ
+from urllib import parse
+
+import aiohttp
+import discord
+from discord.ext import commands
+
+TMDB_API_KEY = environ.get("TMDB_API_KEY")
+
+log = logging.getLogger(__name__)
+
+
+class RomanceMovieFinder:
+    """
+    A cog that returns a random romance movie suggestion to a user
+    """
+
+    def __init__(self, bot):
+        self.bot = bot
+
+    @commands.command(name="romancemovie")
+    async def romance_movie(self, ctx):
+        """
+        Randomly selects a romance movie and displays information about it
+        """
+        # selecting a random int to parse it to the page parameter
+        random_page = random.randint(0, 20)
+        # TMDB api params
+        params = {"api_key": TMDB_API_KEY,
+                  "language": "en-US",
+                  "sort_by": "popularity.desc",
+                  "include_adult": "false",
+                  "include_video": "false",
+                  "page": random_page,
+                  "with_genres": "10749"
+                  }
+        # the api request url
+        request_url = "https://api.themoviedb.org/3/discover/movie?" + parse.urlencode(params)
+        async with aiohttp.ClientSession() as session:
+            async with session.get(request_url) as resp:
+                # loading the json file returned from the api
+                data = await resp.json()
+                # selecting random result from results object in the json file
+                selected_movie = random.choice(data["results"])
```

Comment:
The API response should probably be checked to make sure it was successful, otherwise `data["results"]` is going to throw a `KeyError`.

Reply:
Used try except to fix the issue